ui.frontend/src/view/FormFileInputWidget.js

```diff
@@ -414,10 +414,14 @@
                             // check if file names are valid (ie) there are no control characters in file names
                             isInvalidFileName = isCurrentInvalidFileName = true;
                             inValidNamefileNames = currFileName + "," + inValidNamefileNames;
-                        } else if (file.type) {
-                            let isMatch = this.regexMimeTypeList.some(function (rx) {
-                                return rx.test(file.type);
-                            });
+                        } else{
+                            let isMatch=false;
+                            if (file.type) {
```

Comment:
this will  lead to error in those scenarios where the customer has explicitly allowed such mimetypes to be uploded but browsers are unable to determine the type. 
browsers typically read the extension to determine the mimetype. So, ideally we should fallback to manual mapping if file.type is undefined. in this case, we should do manual mapping of msft specific mimetypes :
`application/x-msdownload` and a`pplication/vnd.ms-outlook `
to the appropriate extensions.
This will still allow users to upload these files by changing to dummy extensions but the behavior is compatible with that of browsers. Strict checking (by first first few bytes of file to sniff mimetype) could be enforced whenever we introduce server side validation but it is outside of scope of this PR.

Reply:
> This will still allow users to upload these files by changing to dummy extensions but the behavior is compatible with that of browsers. Strict checking (by reading first few bytes of file to sniff mimetype) could be enforced whenever we introduce server side validation but it is outside of scope of this PR.

Yes, this is already done on server today for AF v1. Its in backlog for AF v2 based on customer use-case

Reply:
added the changes suggested and the testcases